src/metabase/models/params/chain_filter.clj

```diff
@@ -115,26 +115,21 @@
 
 (defn- filter-clause
   "Generate a single MBQL `:filter` clause for a Field and `value` (or multiple values, if `value` is a collection)."
-  [source-table-id field-id value]
+  [source-table-id {:keys [field-id op value options]
+                    :or   {op :=}}]
   (let [field-clause (let [this-field-table-id (field/field-id->table-id field-id)]
                        [:field field-id (when-not (= this-field-table-id source-table-id)
                                           {:join-alias (joined-table-alias this-field-table-id)})])]
-    (cond
-      ;; e.g. {$$venues.price [:between 2 3]} -> [:between $venues.price 2 3]
-      ;; this is not really supported by the API directly
-      (and (sequential? value) (keyword? (first value)))
-      (into [(first value) field-clause] (rest value))
-      ;; e.g. {$$venues.price #{2 3}} -> [:= $$venues.price 2 3]
-      (and (coll? value) (not (map? value)))
-      (into [:= field-clause] value)
-      :else
-      ;; e.g. {$$venues.price "past32weeks"} -> [:time-interval $checkins.date -32 :week]
-      (or (when (and (temporal-field? field-id)
-                     (string? value))
-            (u/ignore-exceptions
-              (params.dates/date-string->filter value field-id)))
-          ;; e.g. {$$venues.price 2} -> [:= $$venues.price 2]
-          [:= field-clause value]))))
+    (if (and (temporal-field? field-id)
+             (string? value))
+      (u/ignore-exceptions
+        (params.dates/date-string->filter value field-id))
+      (cond-> [op field-clause]
+        true          (into (if (or (sequential? value)
+                                    (set? value))
+                              value
+                              [value]))
+        (seq options) (conj options)))))
```

Comment:
if this has an options, then the query will be sth like `[:!= [:field 443 nil] "American" {:case-sensitive true}]`

which is a invalid mbql.

I don' think this is how options is used

Reply:
Looking at schema, this is exactly how it's used:

```
metabase.mbql.schema> (mc/validate Filter [:contains [:field 443 nil] "American" {:case-sensitive true}])
true
```

I've changed test a little bit (locally, did not push not to trigger CI) so this can be demonstrated:

```
(testing "contains is case insensitive"
          (mt/let-url [url (chain-filter-values-url dashboard (:category-name param-keys)
                                                    (:not-category-name param-keys) "American"
                                                    (:category-contains param-keys) "am")]
            (is (= {:values          [["Latin American"] ["Ramen"]]
                    :has_more_values false}
                   (chain-filter-test/take-n-values 3 (mt/user-http-request :rasta :get 200 url))))))
                   ```

Reply:
oh okay, it was not invalid because I used `:!=` :D

Reply:
hmm, but it's still possible to generate an invalid clause tho... what if options is sth else? the options is currently `(merge (:options ttag) (:options field))` so I have no idea what i could be

how about we make the options strictly to be only `:case-sensitive`?

Reply:
Maybe it's better to leave validation to an MBQL layer so that it's not specified multiple times over the codebase? I guess there is a point in catching in earlier, but OTOH the level of specificity of MBQL schema is really high, so IDK if there is a necessity to put any effort in a much simpler schema?..

Adding any other option would be a pain, since no one will remember you have to add new option to chain filters, and then it won't work. IDK. 🤣